pkg/virt-config/config-map.go

```diff
@@ -150,22 +160,36 @@ func defaultClusterConfig() *Config {
 		EmulatedMachines:       emulatedMachinesDefault,
 		LessPVCSpaceToleration: DefaultLessPVCSpaceToleration,
 		NodeSelectors:          nodeSelectorsDefault,
+		GuestManagementOverheadConfig: &GuestManagementOverheadConfig{
+			VideoMemoryOverhead:    &videoMemoryOverheadDefault,
```

Comment:
I'm not sure there's a reason to break all the overhead into separate variables like this. Isn't all this overhead being aggregated together in the end anyway?  

Why couldn't it just be a single value to represent memory overhead?

Reply:
Good point, I also considered this. Most of the properties are indeed just being aggregated with the exception of (see [getMemoryOverhead](https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-controller/services/template.go#L976)):
1. Overhead per-vCPU that depends on the number of CPUs
2. Video overhead that depends on whether or not the VM is set with a graphic device

Maybe we could say that (1) can be hard-coded (if it is the same on other architectures like power) but I didn't see a way to address (2) so I ended up putting all those things in that strcuts - I'm open for other suggestion though

Reply:
I see, so vcpu memory and video memory make the overhead dynamic depending on the VMI spec. 

How likely is the vcpu or video memory overhead to change over time though? I'd expect the primary variable in overhead to be the kubevirt dependencies running in the container. If we think vcpu and video memory will remain stable, maybe we can hardcode those and only expose the kubevirt dep overhead?

Reply:
Hm, I am not sure if this approach is really the one to go. For instance the video overhead can now be tuned but the video ram is hardcoded. How about just having a factor in cluster-config?

Reply:
> I see, so vcpu memory and video memory make the overhead dynamic depending on the VMI spec.
> 
> How likely is the vcpu or video memory overhead to change over time though? I'd expect the primary variable in overhead to be the kubevirt dependencies running in the container. If we think vcpu and video memory will remain stable, maybe we can hardcode those and only expose the kubevirt dep overhead?

Yeah, that makes sense.
My focus was not on making these values configurable but rather on to enable users to increase the memory overhead not by changing Resources.Limits.Memory and Resources.Requests.Memory, and with this approach, this can be achieved with simpler cluster-level configuration.

> Hm, I am not sure if this approach is really the one to go. For instance the video overhead can now be tuned but the video ram is hardcoded. How about just having a factor in cluster-config?

Sorry but I'm not sure I understand this suggestion - do you mean having a single value, like @davidvossel  suggests, but specifying it as a factor rather than as an absolute value? If so, what would be the base value we will compute the overall memory overhead with, in this case?

Reply:
> Sorry but I'm not sure I understand this suggestion - do you mean having a single value, like @davidvossel suggests, but specifying it as a factor rather than as an absolute value? If so, what would be the base value we will compute the overall memory overhead with, in this case?

What I mean, the values in the formula are not coming from nothing. It is unlikely that a specific value is the issue. Absolute value, or factor does not matter much to me. In the end the main problem will be when people add sidecars which do changes to qemu directly. In this case a cluster-level tune-able will not be the solution, so I am a little bit uncertain what to think of this change.

Reply:
Ah ok, so I really didn't understand your previous message correctly :)

So you're saying: having it configurable at the cluster-level is fundamentally wrong because typically users would have different types of VMIs deployed with various sidecars that lead to different memory overhead on a per-VM basis, right?

Ok, let put this scenario aside for a second and consider two cases:
1. We set an incorrect hard-coded overhead for our kubevirt-level dependencies
2. Users add sidecars to all VMIs in the cluster

Let's not talk about the probability of this to happen, ok? I think it is possible and for those cases, a field that allows increasing (decreasing is not that likely) at the cluster-level can be beneficial I think - do you agree?

I see your point about having different "templates" of VMIs that are deployed with different sidecars - it's just that it doesn't conflict with this cluster-level configuration as long as it would be possible to override it on a per-VMI basis

Reply:
@ahadas I think having one value which will be added can make sense. Does that work for you? Exposing every variable of the formula seems too much to me. Or do we see a special value in that?

Reply:
Yep, exposing one value works for me.

Reply:
Btw, be aware of the fact that some of the overhead depends on external factors. For example qemu needs a buffer for each connected VNC client. Also some of the VM overhead is ratio based and some is fixed. We can let the user specify the number, but I expect the users to trust us here most of the time.

Reply:
Thinking about this, I  believe a better approach would be to give the user an option to turn off the overhead for the VM or to set the max expected number of connected users and so on. And all of that without forcing the user to compute exact numbers. Because frankly, no user will be able to compute that properly as it relies on too many internal and external factors.

Reply:
@MarSik yes, users are normally not expected to set it manually indeed. Interestingly, k8s seems to address the same problem by [introducing a separate field for the overhead that is autopopulated at admission time](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/20190226-pod-overhead.md) as well, but they've added it to the `Spec` - I guess they didn't consider overcommitment.